packaging/amzn2/snapd.spec

```diff
@@ -0,0 +1,299 @@
+%global repo            snapd
+# https://github.com/snapcore/snapd
+%global provider_prefix %{provider}.%{provider_tld}/%{project}/%{repo}
+%global import_path     %{provider_prefix}
+
+%define systemd_services_list snapd.socket snapd.service snapd.autoimport.service
+
+%define gobuild(o:) go build -compiler gc -tags=rpm_crashtraceback -ldflags "${LDFLAGS:-} -B ${BUILDID:-0x0} -extldflags '%__global_ldflags'" -a -v -x %{?**};
+%define gobuild_static(o:) go build -compiler gc -tags=rpm_crashtraceback -ldflags "${LDFLAGS:-} -B ${BUILDID:-0x0} -extldflags '%__global_ldflags -static'" -a -v -x %{?**};
+%define gotest() go test -compiler gc -ldflags "${LDFLAGS:-}" %{?**};
+
+Name:           snapd
+Version:        2.32.3
+Release:        0%{?dist}
+Summary:        Tools enabling systems to work with .snap files
+License:        GPLv3
+Group:          System Environment/Base
+Url:            https://%{provider_prefix}
+# Use the version with the vendored go packages
+Source0:  https://%{provider_prefix}/releases/download/%{version}/%{name}_%{version}.vendor.tar.xz
+BuildRequires:  autoconf
+BuildRequires:  automake
+BuildRequires:  gcc
+BuildRequires:  gettext
+BuildRequires:  gnupg
+BuildRequires:  pkgconfig(glib-2.0)
+BuildRequires:  pkgconfig(libcap)
+BuildRequires:  pkgconfig(libseccomp)
+BuildRequires:  pkgconfig(libudev)
+BuildRequires:  pkgconfig(systemd)
+BuildRequires:  libseccomp-static
+BuildRequires:  glibc-static
+BuildRequires:  golang
+BuildRequires:  libtool
+BuildRequires:  make
+BuildRequires:  pkgconfig
+BuildRequires:  squashfs-tools
+BuildRequires:  xfsprogs-devel
+BuildRequires:  xz
+BuildRequires:  valgrind
+
+%{?systemd_requires}
+Requires: squashfs-tools
+# TODO: enable squashfuse once it's packaged
+#Requires:  squashfuse
+#Requires:  fuse
+Requires: bash-completion
+
+BuildRoot:      %{_tmppath}/%{name}-%{version}-build
+
+%description
+This package contains that snapd daemon and the snap command line tool.
+Together they can be used to install, refresh (update), remove and configure
+snap packages on a system. Snap packages are a novel format based on simple
+principles. Bundle your dependencies, run in a predictable environment, use
+moder kernel features for setting up the execution environment and security.
+The same binary snap package can be installed and used on many diverse systems
+such as Debian, Fedora and OpenSUSE as well as their multiple derivatives.
+
+%prep
+%setup -q -n %{name}-%{version}
+
+%build
+# Set the version that is compiled into the various executables
+./mkversion.sh %{version}-%{release}
+
+# Build golang executables
+mkdir -p src/github.com/snapcore
+ln -s ../../../ src/github.com/snapcore/snapd
+
+export GOPATH=$(pwd):$(pwd)/Godeps/_workspace:%{gopath}
+
+GOFLAGS=
+%if 0%{?with_test_keys}
+GOFLAGS="$GOFLAGS -tags withtestkeys"
+%endif
+
+BUILDID="0x$(head -c20 /dev/urandom|od -An -tx1|tr -d ' \\\n')"
+
+%gobuild -o bin/snapd $GOFLAGS github.com/snapcore/snapd/cmd/snapd
```

Comment:
We need those because of missing packaging helpers in Amazon linux?

Reply:
Yup, their macros for go is literally just this:
```
[ec2-user@localhost rpm]$ cat /usr/lib/rpm/macros.d/macros.golang 
# Where to set GOPATH for builds. Like:
# export GOPATH=$(pwd)/_build:%{gopath}
%gopath          %{_datadir}/gocode

# for use like:
# ExclusiveArch:  %{go_arches}
%go_arches       x86_64 aarch64 ppc64le s390x

```

Reply:
It's because CentOS 7 (of which Amazon Linux 2 is derived from) has no useful Go macros. That's why I inlined them into the Fedora spec for EL7 support.